StripePaymentSheet/StripePaymentSheet/Source/Helpers/SavedPaymentMethodManager.swift

```diff
@@ -6,26 +6,52 @@
 //
 
 import Foundation
+@_spi(STP) import StripeCore
 @_spi(STP) import StripePayments
 
 /// Provides shared implementations of common operations for managing saved payment methods in PaymentSheet
 final class SavedPaymentMethodManager {
 
+    enum Error: Swift.Error {
+        case missingEphemeralKey
+    }
+
     let configuration: PaymentSheet.Configuration
+    let intent: Intent
+
+    private var ephemeralKey: String? {
+        guard let ephemeralKey = configuration.customer?.ephemeralKeySecretBasedOn(intent: intent) else {
+            stpAssert(true, "Failed to read ephemeral key.")
+            let errorAnalytic = ErrorAnalytic(event: .unexpectedPaymentSheetError,
+                                              error: Error.missingEphemeralKey,
+                                              additionalNonPIIParams: ["customer_access_provider": configuration.customer?.customerAccessProvider.analyticValue as Any])
```

Comment:
```suggestion
                                              additionalNonPIIParams: ["customer_access_provider": configuration.customer?.customerAccessProvider.analyticValue as String])
```

nit: Does that not work?

Reply:
It doesn't since it's optional, we could unwrap it instead I suppose.
![CleanShot 2024-07-03 at 13 52 29](https://github.com/stripe/stripe-ios/assets/88012362/45c19aa0-e123-4447-8f8a-14f999be6776)

Reply:
`as String?`? Or provide a default value?  I'm just slightly wary of `Any`

Reply:
Yeah similar warning, I'll provide a default value.
![CleanShot 2024-07-03 at 14 01 16](https://github.com/stripe/stripe-ios/assets/88012362/0fb5ea05-ab66-4ba4-9793-8a10fb8b1e99)